test/test_epoll.cpp

```diff
@@ -555,3 +556,210 @@ TEST(CEPoll, ThreadedUpdate)
+                        read,  &rlen,
+                        write, &wlen,
+                        -1, // -1 is set for debuging purpose.
+                        // in case of production we need to set appropriate value
+                        0, 0, 0, 0), SRT_ERROR );
+
+            ASSERT_EQ(rlen, 1); // get exactly one read event without writes
+            ASSERT_EQ(wlen, 0); // get exactly one read event without writes
+            ASSERT_EQ(read[0], servsock); // read event is for bind socket    	
+        }
+
+        sockaddr_in scl;
+        int sclen = sizeof scl;
+
+        SRTSOCKET sock = srt_accept(servsock, (sockaddr*)&scl, &sclen);
+        ASSERT_NE(sock, SRT_INVALID_SOCK);
+
+        srt_epoll_add_usock(m_server_pollid, sock, &epoll_in); // wait for input
+
+        { // wait for 1316 packet from client
+            int rlen = 2;
+            SRTSOCKET read[2];
+
+            int wlen = 2;
+            SRTSOCKET write[2];
+
+            ASSERT_NE(srt_epoll_wait(m_server_pollid,
+                        read,  &rlen,
+                        write, &wlen,
+                        -1, // -1 is set for debuging purpose.
+                        // in case of production we need to set appropriate value
+                        0, 0, 0, 0), SRT_ERROR );
+
+            ASSERT_EQ(rlen, 1); // get exactly one read event without writes
+            ASSERT_EQ(wlen, 0); // get exactly one read event without writes
+            ASSERT_EQ(read[0], sock); // read event is for bind socket        
+        }
+
+        char buffer[1316];
+        ASSERT_EQ(srt_recvmsg(sock, buffer, sizeof buffer), 1316);
+
+        char pattern[4] = {1, 2, 3, 4};
+        EXPECT_TRUE(std::mismatch(pattern, pattern+4, buffer).first == pattern+4);
+
+        std::cout << "serverSocket waiting..." << std::endl;
+        {
+            int rlen = 2;
+            SRTSOCKET read[2];
+
+            int wlen = 2;
+            SRTSOCKET write[2];
+
+            ASSERT_EQ(-1, srt_epoll_wait(m_server_pollid,
+                        read,  &rlen,
+                        write, &wlen,
+                        2000,
+                        0, 0, 0, 0));
+            const int last_error = srt_getlasterror(NULL);
+            ASSERT_EQ(SRT_ETIMEOUT, last_error) << last_error;
+        }
+        std::cout << "serverSocket finished waiting" << std::endl;
+
+        srt_close(sock);
+        srt_close(servsock);
+        srt_close(m_client_sock); // cannot close m_client_sock after srt_sendmsg because of issue in api.c:2346 
+
+        client.join();
+    }
+
+};
+
+
+TEST_F(TestEPoll, SimpleAsync)
+{
+    ASSERT_EQ(srt_startup(), 0);
+
+    m_client_pollid = srt_epoll_create();
+    ASSERT_NE(SRT_ERROR, m_client_pollid);
+
+    m_server_pollid = srt_epoll_create();
+    ASSERT_NE(SRT_ERROR, m_server_pollid);
+
+    serverSocket();
+
+    (void)srt_epoll_release(m_client_pollid);
+    (void)srt_epoll_release(m_server_pollid);
+    srt_cleanup();
```

Comment:
Move release et al to `void TestEPoll::TearDown() override`. It is called by GTest just after the test completes.